File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.kt
class Solution {
    fun solution(k: Int, score: IntArray): IntArray {
        val list = mutableListOf<Int>()
        val answer = IntArray(score.size) { 0 }

        for(i in score.indices) {
            list.add(score[i])
            list.sort()
            if(list.size > k) {
                list.removeAt(0)
            }
            answer[i] = list[0]
        }
        return answer
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.swift
func solution(_ k:Int, _ score:[Int]) -> [Int] {
    var answer = Array<Int>()
    var accumulationScore = Array<Int>()
    
    for s in score {
        accumulationScore.append(s)
        accumulationScore.sort(by: >)
        
        let lowestRank = getLowestRank(k, accumulationScore)
        answer.append(lowestRank)
    }
    
    return answer
}

func getLowestRank(_ k: Int, _ scores: [Int]) -> Int {
    return scores.count <= k ? scores[scores.count-1] : scores[k-1]
}